src/plugins/workspace/server/workspace_client.ts

```diff
@@ -65,6 +79,94 @@ export class WorkspaceClientWithSavedObject implements IWorkspaceDBImpl {
   private formatError(error: Error | any): string {
     return error.message || error.error || 'Error';
   }
+  private async checkAndCreateWorkspace(
+    savedObjectClient: SavedObjectsClientContract | undefined,
+    workspaceId: string,
+    workspaceAttribute: Omit<WorkspaceAttribute, 'id' | 'permissions'>,
+    permissions?: Permissions
+  ) {
+    try {
+      await savedObjectClient?.get(WORKSPACE_TYPE, workspaceId);
+    } catch (error) {
+      this.logger.debug(error?.toString() || '');
+      this.logger.info(`Workspace ${workspaceId} is not found, create it by using internal user`);
+      try {
+        const createResult = await savedObjectClient?.create(WORKSPACE_TYPE, workspaceAttribute, {
+          id: workspaceId,
+          permissions,
+        });
+        if (createResult?.id) {
+          this.logger.info(`Created workspace ${createResult.id}.`);
+        }
+      } catch (e) {
+        this.logger.error(`Create ${workspaceId} workspace error: ${e?.toString() || ''}`);
+      }
+    }
+  }
+  private async setupPublicWorkspace(savedObjectClient?: SavedObjectsClientContract) {
+    const publicWorkspaceACL = new ACL().addPermission([WorkspacePermissionMode.Management], {
+      users: ['*'],
+    });
+    return this.checkAndCreateWorkspace(
+      savedObjectClient,
+      PUBLIC_WORKSPACE_ID,
+      {
+        name: i18n.translate('workspaces.public.workspace.default.name', {
+          defaultMessage: 'public',
+        }),
+        features: ['*', `!@${DEFAULT_APP_CATEGORIES.management.id}`],
+        reserved: true,
+      },
+      publicWorkspaceACL.getPermissions()
+    );
+  }
+  private async setupManagementWorkspace(savedObjectClient?: SavedObjectsClientContract) {
+    const managementWorkspaceACL = new ACL().addPermission([WorkspacePermissionMode.Management], {
+      users: ['*'],
+    });
```

Comment:
For management workspace, are we going to grant `WorkspacePermissionMode.Management` to all users?

Reply:
Yes, the conclusion is that we will make public and management workspace manageable by anyone, and the first one can change the permission if they want to limit the permission.